packages/nodes-base/nodes/If/test/v2/IfV2.node.test.ts

```diff
@@ -1,5 +1,155 @@
+
+			jest.spyOn(IfV2, 'getOptions').mockImplementation(() => {
+				throw new NodeOperationError(mock(), 'Node Operation Error');
+			});
+
+			await expect(node.execute.call(thisArg)).rejects.toThrow(NodeOperationError);
+		});
+
+		it('should throw an error if a ApplicationError  is thrown and set the context of the error', async () => {
+			thisArg.getInputData.mockReturnValue([{ json: {} }]);
+
+			const error = new ApplicationError('Error');
+
+			jest.spyOn(IfV2, 'getOptions').mockImplementation(() => {
+				throw error;
+			});
+
+			await expect(node.execute.call(thisArg)).rejects.toThrow(ApplicationError);
+			expect(set).toHaveBeenCalledWith(error, 'context.itemIndex', 0);
+		});
+
+		it('should throw a NodeOperationError error if the error is not NodeOperationError of Application Error', async () => {
+			thisArg.getInputData.mockReturnValue([{ json: {} }]);
+
+			const error = new Error('Error');
+
+			jest.spyOn(IfV2, 'getOptions').mockImplementation(() => {
+				throw error;
+			});
+
+			await expect(node.execute.call(thisArg)).rejects.toThrow(NodeOperationError);
+		});
+
+		it('should throw a NodeOperationError and set description of error if error thrown while getting conditions and if error description is not set', async () => {
+			const error = new Error();
+			const mockNode = mock<INode>();
+
+			thisArg.getInputData.mockReturnValue([{ json: {} }]);
+			thisArg.getNodeParameter
+				.calledWith('options', 0)
+				.mockReturnValue({ looseTypeValidation: false });
+			thisArg.getNode.calledWith().mockReturnValue(mockNode);
+
+			jest.spyOn(IfV2, 'getConditions').mockImplementation(() => {
+				throw error;
+			});
+
+			jest.spyOn(IfV2Utils, 'getTypeValidationParameter').mockReturnValue(() => false);
+
+			await expect(node.execute.call(thisArg)).rejects.toThrow(NodeOperationError);
+
+			expect(set).toHaveBeenCalledWith(error, 'description', ENABLE_LESS_STRICT_TYPE_VALIDATION);
+			expect(set).toHaveBeenCalledWith(error, 'context.itemIndex', 0);
+			expect(set).toHaveBeenCalledWith(error, 'node', mockNode);
+		});
+
+		it('should assign a paired item if paired item is undefined', async () => {
+			const mockNode = mock<INode>();
+
+			thisArg.getInputData.mockReturnValue([{ json: {} }]);
+			thisArg.getNodeParameter
+				.calledWith('options', 0)
+				.mockReturnValue({ looseTypeValidation: false });
+			thisArg.getNodeParameter
+				.calledWith('condition', 0, false, { extractValue: true })
+				.mockReturnValue('false');
+			thisArg.getNode.calledWith().mockReturnValue(mockNode);
+
+			jest.spyOn(IfV2Utils, 'getTypeValidationParameter').mockReturnValue(() => false);
+
+			const output = await node.execute.call(thisArg);
+			expect(output).toEqual([[], [{ json: {}, pairedItem: { item: 0 } }]]);
+		});
+	});
+
+	describe('Test IF V2 Node Util Tests', () => {
+		const thisArg = mock<IExecuteFunctions>({});
+
+		it('should return an option if the node type version is less than the version passed in', () => {
```

Comment:
I like how you are testing that looseTypeValidation has a different default value based on the node version 👍